pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2182,13 +2182,17 @@ protected ResultSet getPrimaryUniqueKeys(@Nullable String catalog, @Nullable Str
     sql = "SELECT NULL AS TABLE_CAT, n.nspname AS TABLE_SCHEM, "
         + "  ct.relname AS TABLE_NAME, a.attname AS COLUMN_NAME, "
         + "  (information_schema._pg_expandarray(i.indkey)).n AS KEY_SEQ, ci.relname AS PK_NAME, "
-        + "  information_schema._pg_expandarray(i.indkey) AS KEYS, a.attnum AS A_ATTNUM "
+        + "  information_schema._pg_expandarray(i.indkey) AS KEYS, a.attnum AS A_ATTNUM, "
+        + "  a.attnotnull AS IS_NOT_NULL "
         + "FROM pg_catalog.pg_class ct "
         + "  JOIN pg_catalog.pg_attribute a ON (ct.oid = a.attrelid) "
         + "  JOIN pg_catalog.pg_namespace n ON (ct.relnamespace = n.oid) "
         + "  JOIN pg_catalog.pg_index i ON ( a.attrelid = i.indrelid) "
         + "  JOIN pg_catalog.pg_class ci ON (ci.oid = i.indexrelid) "
-        + "WHERE true ";
+        // primary as well as unique keys can be used to uniquely identify a row to update
+        + "WHERE (i.indisprimary or i.indisunique) "
```

Comment:
@davecramer There is a way to have the index already filtered by all not null attributes (see below) but not sure if this would be better in terms of performance - would need to have primary check OR (all-unique-checks-together)
```sql
AND NOT EXISTS (
  SELECT 1
  FROM pg_catalog.pg_attribute ai
  WHERE ai.attrelid = i.indrelid
    AND ai.attnotnull = FALSE
) 
```

this will remove the need to read the IS_NOT_NULL property for all columns and check in the while(rs.next()) loop on PgResultSet

I have the patch available if needed / preferred this way